rendermodules/point_match_optimization/point_match_optimization.py

```diff
@@ -0,0 +1,313 @@
+import os
+import json
+import renderapi
+import itertools
+import numpy as np
+import marshmallow as mm
+import cv2
+import matplotlib.pyplot as plt
+import matplotlib.image as mpimg
+import tempfile
+import time
+from numbers import Number
+from functools import partial
+from marshmallow import Schema, fields
+from jinja2 import FileSystemLoader, Environment
+from rendermodules.module.render_module import RenderModule
+from rendermodules.point_match_optimization.schemas import PointMatchOptimizationParameters, PointMatchOptimizationParametersOutput
+
+
+ex = {
+    "render": {
+        "host": "http://em-131fs",
+        "port": 8080,
+        "owner": "gayathri",
+        "project": "MM2",
+        "client_scripts": "/allen/programs/celltypes/workgroups/em-connectomics/gayathrim/nc-em2/Janelia_Pipeline/render_latest/render-ws-java-client/src/main/scripts"
+    },
+    "SIFT_options": {
+        "SIFTfdSize": [8],
+        "SIFTmaxScale": [0.82],
+        "SIFTminScale": [0.28],
+        "SIFTsteps": [3],
+        "renderScale": [0.1, 0.35]
+    },
+    "outputDirectory": "/allen/programs/celltypes/workgroups/em-connectomics/gayathrim/scratch/pmopts",
+    "tile_stack": "point_match_optimization_test",
+    "stack": "mm2_acquire_8bit_reimage",
+    "tileId1": "20170830120417214_295434_5LC_0064_01_redo_001050_0_16_12.1050.0",
+    "tileId2": "20170830120445578_295434_5LC_0064_01_redo_001050_0_16_13.1050.0",
+    "url_options":{
+        "normalizeForMatching": "true",
+        "renderWithFilter": "true",
+        "renderWithoutMask": "true"
+    }
+}
+
+def get_canvas_url(render_args, stack, tile1, tile2, url_options):
+    base_data_url = '%s:%d/render-ws/v1'%(render_args['host'], render_args['port'])
+    tile_base_url = "%s/owner/%s/project/%s/stack/%s/tile"%(base_data_url,
+                                                            render_args['owner'],
+                                                            render_args['project'],
+                                                            stack)
+    url_suffix = "render-parameters"
+    if url_options['renderWithFilter'] is False:
+        url_suffix += '?filter=false'
+    else:
+        url_suffix += '?filter=true'
+    if url_options['normalizeForMatching'] is False:
```

Comment:
I think we should switch to using the newer transform labels and removing transform labels, rather than normalizeForMatching=true.  If in this case we want to excludeAllTransforms=true, or excludeTransformsAfterLast

Reply:
Agreed, but in this code it should be in addition to the normalizeForMatching functionality.  I think in the current implementation spark-based SIFT matching will assume normalizeForMatching=True.

Reply:
The current spark sift client runs on a tilepair file that contains the render-parameters template url, if alternate keyword arguments are given then the tilespecs returned will work.. what it assumes is that the tilespecs will be altered such that the image will be around 0,0 and the point matches will be extracted from those images as 'local' coordinates.

Reply:
that said i have no argument leaving it as an option.  We should also double back to the apply lens correction module and make sure there is an option to label the lens correction transform with a transform label, with a default of 'lens'.

Reply:
ok.. i think we won't add this for this release and we can come back later when the lens labels are included and we have fixed the spark client.